src/main/java/sirius/biz/jobs/params/MultiSelectStringParameter.java

```diff
@@ -0,0 +1,140 @@
+     */
+    public MultiSelectStringParameter(String name, String label) {
+        super(name, label);
+    }
+
+    /**
+     * Adds an entry to the list.
+     *
+     * @param key   the entry key
+     * @param value the display value, which will be {@link NLS#smartGet(String) auto translated before display}
+     * @return the parameter itself for fluent method calls
+     */
+    public MultiSelectStringParameter withEntry(String key, String value) {
+        if (this.entriesProvider != null) {
+            throw new IllegalStateException("Entries can not be added when an entries provider is set.");
+        }
+
+        this.entries.put(key, value);
+        return self();
+    }
+
+    /**
+     * Sets a provider that generates the collection of selectable entries.
+     * <p>
+     * Use this to lazily initialize the entries. A {@link CachingSupplier} can be used to cache the entries.
+     *
+     * @param entriesProvider the provider that returns list of entries
+     * @return the parameter itself for fluent method calls
+     */
+    public MultiSelectStringParameter withEntriesProvider(Supplier<Map<String, String>> entriesProvider) {
+        if (!entries.isEmpty()) {
+            throw new IllegalStateException("An entry provider can not be set after entries have already been added.");
+        }
+
+        this.entriesProvider = entriesProvider;
+        return self();
+    }
+
+    private Map<String, String> fetchEntriesMap() {
+        if (entriesProvider != null) {
+            return entriesProvider.get();
+        }
+        return entries;
+    }
+
+    /**
+     * Enumerates all values provided by the parameter.
+     *
+     * @return list of {@linkplain Tuple entries} with the key as first and display value as second tuple items.
+     */
+    @Override
+    public List<Tuple<String, String>> getValues() {
+        return fetchEntriesMap().keySet()
+                                .stream()
+                                .map(entry -> Tuple.create(entry, NLS.smartGet(fetchEntriesMap().get(entry))))
+                                .toList();
+    }
+
+    @Override
+    protected String checkAndTransformValue(Value input) {
+        if (!(input.get() instanceof List<?> list)) {
+            return checkAndTransformSingleValue(input);
+        }
+
+        String verifiedInput = list.stream()
+                                   .map(Value::of)
+                                   .map(this::checkAndTransformSingleValue)
+                                   .filter(Objects::nonNull)
+                                   .collect(Collectors.joining(","));
+        return Strings.isFilled(verifiedInput) ? verifiedInput : null;
+    }
+
+    private String checkAndTransformSingleValue(Value input) {
+        String rawInput = input.asString().trim();
+
+        // we use the comma to separate the values, and thus can not allow it within values
+        if (rawInput.contains(",")) {
```

Comment:
**Optional:** Maybe switch to `|` as a separator which is far less common to be included in entries and also understood by CSVFilter of sirius-db

Reply:
Refactored code to move delimiter into a constant field & changed value to "|"